sdcm/remote.py

```diff
@@ -963,3 +967,126 @@ def run_output_check(self, cmd, timeout=None, ignore_status=False,
 
         if not ignore_status and result.exit_status > 0:
             raise process.CmdError(command=result.command, result=result)
+
+
+class RemoteFabric(object):
+
+    # def __init__(self, hostname, user="root", port=22, password="",
+                 # key_file=None, wait_key_installed=0, extra_ssh_options=""):
+    def __init__(self, *args, **kwargs):
+        logger = logging.getLogger('avocado.test')
+        self.log = SDCMAdapter(logger, extra={'prefix': str(self)})
+        self.log.info(args)
+        self.log.info(kwargs)
+        self.key_file = kwargs.get('key_file', "")
+        self.user = kwargs.get('user', "scylla-test")
+        self.port = kwargs.get('port', 22)
+        self.hostname = kwargs.get('hostname', "localhost")
+        self.password = kwargs.get('password', "")
+        self.connect_timeout = kwargs.get('connect_timeout', 60)
+        self.cmd_timeout = kwargs.get('timeout', 300)
+        self.known_hosts_file = tempfile.mkstemp()[1]
+        config = Config(overrides={
+                        'load_ssh_config': False,
+                        'UserKnownHostsFile': self.known_hosts_file})
+        self.connection = Connection(
+            self.hostname,
+            self.user,
+            port=self.port,
+            config=config,
+            connect_timeout=self.connect_timeout,
+            connect_kwargs={
+                'key_filename': os.path.expanduser(self.key_file),
+            })
+        self._check_install_key_required()
+
+    def ssh_debug_cmd(self):
+        self.log.info("{0.key_file} {0.hostname} {0.user}".format(self))
+
+    def run(self, cmd, *args, **kwargs):
+        self.log.info('Running command {}'.format(cmd))
+        self.log.info("args {}, {}".format(args, kwargs))
+        result = self.connection.run(cmd, warn=True, encoding='utf-8')
```

Comment:
I remember that the default behavior of fabric was to echo the command and the stdout/stderr, so printing all those with logger might be duplication of prints, maybe the hide=True, needs to be used:
http://docs.pyinvoke.org/en/1.2/api/runners.html#invoke.runners.Runner.run

Reply:
Yes, i used it to get all possible parameters, which are passed to run method.

Reply:
Will update